src/main/java/com/rk/rentacarapp/utils/JwtUtil.java

```diff
@@ -0,0 +1,144 @@
+package com.rk.rentacarapp.utils;
+
+import com.rk.rentacarapp.model.User;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Date;
+
+@Component
+public class JwtUtil {
+
+    private String APP_NAME = "Rent a car application";
+
+    public String SECRET = "ourLittleSecret, psst!";
+
+    private int EXPIRES_IN = 180; //Value represents minutes (so.. 3h)
+
+    private String AUTH_HEADER = "Authorization";
+
+    private SignatureAlgorithm SIGNATURE_ALGORITHM = SignatureAlgorithm.HS256;
+
+    // Method for generating new JWT token
+
+    public String generateToken(UserDetails userDetails) {
+        return Jwts.builder()
+                .setIssuer(APP_NAME)
+                .setSubject(userDetails.getUsername())
+                .setIssuedAt(new Date())
+                .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * EXPIRES_IN))
+                .signWith(SIGNATURE_ALGORITHM, SECRET).compact();
+    }
+
+    // Method for refreshing JWT token
+
+    public String refreshToken(String token) {
+        String refreshedToken;
+        try {
+            final Claims claims = this.getAllClaimsFromToken(token);
+            claims.setIssuedAt(new Date());
+            refreshedToken = Jwts.builder()
+                    .setClaims(claims)
+                    .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * EXPIRES_IN))
+                    .signWith(SIGNATURE_ALGORITHM, SECRET).compact();
+        } catch (Exception e) {
+            refreshedToken = null;
+        }
+        return refreshedToken;
+    }
+
+    public Boolean canTokenBeRefreshed(String token, Date lastPasswordReset) {
+        final Date created = this.getIssuedAtDateFromToken(token);
+        return (!(this.isCreatedBeforeLastPasswordReset(created, lastPasswordReset))
+                && (!(this.isTokenExpired(token))));
+    }
+
+    // Methods for validating JWT token data
+
+    public Boolean validateToken(String token, UserDetails userDetails) {
+        User user = (User) userDetails;
+        final String username = getUsernameFromToken(token);
+        final Date created = getIssuedAtDateFromToken(token);
+
+        return (username != null && username.equals(userDetails.getUsername())
+                && !isCreatedBeforeLastPasswordReset(created, user.getLastPasswordResetDate()));
+    }
+
+    private Boolean isCreatedBeforeLastPasswordReset(Date created, Date lastPasswordReset) {
```

Comment:
createdAt would be a better variable name, created sounds non defined somehow

Reply:
I'm not sure of what's the usage of this method, but maybe it would make sense to flip the naming as of the one below and call it isCreatedAfterLastPasswordReset. But I'm not sure :)